include/cvc5/cvc5_proof_rule.h

```diff
@@ -169,9 +169,30 @@ enum ENUM(ProofRule)
    * :math:`\texttt{Evalutor::evaluate}` in :cvc5src:`theory/evaluator.h` with an
    * empty substitution.
    * Note this is equivalent to: ``(REWRITE t MethodId::RW_EVALUATE)``.
+   *
+   * Note this proof rule only applies to atomic sorts, that is, operators on
+   * Int, Real, String, Bool or BitVector.
    * \endverbatim
    */
   EVALUE(EVALUATE),
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Builtin theory -- Distinct values**
+   *
+   * .. math::
+   *   \inferrule{- \mid t, s}{\neg t = s}
+   *
+   * where :math:`t` and :math:`s` are distinct values.
+   *
+   * Internally, this means the Node::isConst method returns true for both,
+   * indicating that are canonical representations of values of their sort.
+   *
+   * In practice, this rule is typically only used to terms of non-atomic
+   * sort, e.g. Sets, Sequences, Datatypes, Arrays, etc.
+   *
+   * \endverbatim
+   */
+  EVALUE(DISTINCT_VALUES),
```

Comment:
Weren't we going to name this something more descriptive, to avoid confusion?

Reply:
I'd prefer a concise name. Do you have an alternative?